File: Data Structures & Algorithms/validate-parentheses/submission-2.cpp
class Solution {
public:
    bool isValid(string s) {
        // Map closing brackets to opening brackets
        std::unordered_map<char, char> mapping = {
            {')', '('}, 
            {']', '['}, 
            {'}', '{'}
        };
        
        std::stack<char> st;
        
        for (char c : s) {
            // If it is a closing bracket
            if (mapping.count(c)) {
                if (st.empty() || st.top() != mapping[c]) {
                    return false;
                }
                st.pop();
            } else {
                // If it is an opening bracket
                st.push(c);
            }
        }
        
        return st.empty();
    }
};

File: Data Structures & Algorithms/palindromic-substrings/submission-2.cs
public class Solution {
    public int CountSubstrings(string s) {
        int count = 0, n = s.Length;
        bool[,] dp = new bool[n, n];
        for (int r = 0; r < n; r++) {
            for (int c = 0; c < n; c++) {
                dp[r, c] = false;
            }
        }
        
        for (int l = n-1; l >= 0; --l) {
            for (int r = l; r < n; ++r) {
                if (s[l] == s[r] && (r - l <= 2 || dp[l+1, r-1])) {
                    dp[l, r] = true;
                    count++;
                }
            }
        }
        return count;
    }
}
